tests/decorators/tests.py

```diff
@@ -434,3 +436,283 @@ def method(self):
         Test().method()
         self.assertEqual(func_name, "method")
         self.assertIsNotNone(func_module)
+
+
+def async_simple_dec_m(func):
+    @wraps(func)
+    async def wrapper(*args, **kwargs):
+        return await func(*args, **kwargs)
+
+    return wrapper
+
+
+def async_myattr_dec_m(func):
+    func.myattr = True
+    return func
+
+
+def async_myattr2_dec_m(func):
+    func.myattr2 = True
+    return func
+
+
+class AsyncMethodDecoratorTests(SimpleTestCase):
+    """
+    Tests for async method_decorator
+    """
+
+    async def test_preserve_signature(self):
+        class Test:
+            @async_simple_dec_m
+            async def say(self, arg):
+                return arg
+
+        result = await Test().say("hello")
+        self.assertEqual("hello", result)
+
+    async def test_preserve_attributes(self):
```

Comment:
These tests don't need to be `async def` unless you're actually using `await` in the test scope. That's true for `test_preserve_signature` above, but not for this test, or most/all that follow, where you're just checking wrapping of async methods, but not actually running them.

Reply:
I have made the required change please review.